File: mpp/lab8/client_java/src/main/java/rest/ArbitersResource.java
package rest;

import domain.Arbiter;

import javax.ws.rs.client.WebTarget;
import javax.ws.rs.core.GenericType;
import javax.ws.rs.core.MediaType;
import java.util.List;

public class ArbitersResource {
    private final API api;

    public ArbitersResource(API api) {
        this.api = api;
    }

    public List<Arbiter> find() {
        WebTarget target = api.getTarget().path("/arbiters");
        return target
                .request(MediaType.APPLICATION_JSON_TYPE)
                .get(new GenericType<>() {
                });
    }
}

File: mpp/lab8/client_csharp/Rest/ArbitersResource.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using client_csharp.Domain;
using Newtonsoft.Json;

namespace client_csharp.Rest
{
    public class ArbitersResource
    {
        private readonly API _api;

        public ArbitersResource(API api)
        {
            _api = api;
        }

        public async Task<List<Arbiter>> Find()
        {
            var response = await _api.Client.GetAsync("arbiters");
            if (!response.IsSuccessStatusCode) return null;
            var responseString = await response.Content.ReadAsStringAsync();
            return JsonConvert.DeserializeObject<List<Arbiter>>(responseString);
        }
    }
}
